components/metrics/lib.rs

```diff
@@ -81,14 +282,15 @@ pub struct PaintTimeMetrics {
 }
 
 impl PaintTimeMetrics {
-    pub fn new(pipeline_id: PipelineId,
-               time_profiler_chan: ProfilerChan,
-               constellation_chan: IpcSender<LayoutMsg>,
-               script_chan: IpcSender<ConstellationControlMsg>)
-        -> PaintTimeMetrics {
+    pub fn new(
+        pipeline_id: PipelineId,
+        time_profiler_chan: ProfilerChan,
+        constellation_chan: IpcSender<LayoutMsg>,
+        script_chan: IpcSender<ConstellationControlMsg>,
+    ) -> PaintTimeMetrics {
```

Comment:
If we add navigation_start as an argument, we can remove the Cell and Option from the member.

Reply:
as discussed in irc: i think we have to keep set_navigation_start for paint time metrics. they're created in pipeline.rs and script_thread.rs, but navigation_start is set in layout_thread/lib.rs